Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -1127,6 +1132,93 @@ lemma map_comp (h : M →ₗ[R] N) : (map S f g h) ∘ₗ f = g ∘ₗ h :=
 lemma map_apply (h : M →ₗ[R] N) (x) : map S f g h (f x) = g (h x) :=
   lift_apply S f (g ∘ₗ h) (IsLocalizedModule.map_units g) x
 
+open LocalizedModule LinearEquiv LinearMap
+
+variable (M)
+
+/-- The linear map `(LocalizedModule S M) → (LocalizedModule S M)` from `iso` is the identity. -/
+lemma iso_localizedModule_eq_refl : iso S (mkLinearMap S M) = refl R (LocalizedModule S M) := by
+  let f := mkLinearMap S M
+  obtain ⟨e, _, univ⟩ := is_universal S f f (map_units f)
+  rw [← toLinearMap_inj, univ (iso S f) ((eq_toLinearMap_symm_comp f f).1 (iso_symm_comp S f).symm)]
+  exact Eq.symm <| univ (refl R (LocalizedModule S M)) (by simp)
+
+variable {M₀ M₀'} [AddCommGroup M₀] [AddCommGroup M₀'] [Module R M₀] [Module R M₀']
+variable (f₀ : M₀  →ₗ[R] M₀') [IsLocalizedModule S f₀]
+variable {M₁ M₁'} [AddCommGroup M₁] [AddCommGroup M₁'] [Module R M₁] [Module R M₁']
+variable (f₁ : M₁ →ₗ[R] M₁') [IsLocalizedModule S f₁]
+
+/-- Formula for `IsLocalizedModule.map` when each localized module is a `LocalizedModule`.-/
+lemma map_LocalizedModules (g : M₀ →ₗ[R] M₁) (m : M₀) (s : S) :
+    ((map S (mkLinearMap S M₀) (mkLinearMap S M₁)) g)
+    (LocalizedModule.mk m s) = LocalizedModule.mk (g m) s := by
+  have := (iso_apply_mk S (mkLinearMap S M₁) (g m) s).symm
+  rw [iso_localizedModule_eq_refl, refl_apply] at this
+  simpa [map, lift, iso_localizedModule_eq_refl S M₀]
+
+lemma map_iso_commute (g : M₀ →ₗ[R] M₁) : (map S f₀ f₁) g ∘ₗ ↑(iso S f₀) =
+    (iso S f₁) ∘ₗ (map S (mkLinearMap S M₀) (mkLinearMap S M₁)) g := by
+  ext x
+  refine induction_on (fun m s ↦ ?_) x
+  have hs : IsUnit ((algebraMap R (Module.End R M₁')) ↑s) := map_units f₁ s
+  have : Function.Injective (hs.unit : Module.End R M₁') := by
+    refine Function.injective_iff_hasLeftInverse.2 ⟨hs.unit.inv, fun x ↦ ?_⟩
+    change (hs.unit⁻¹ * hs.unit.1) x = x
+    rw [IsUnit.unit_spec, IsUnit.val_inv_mul, one_apply]
+  apply this
+  rw [IsUnit.unit_spec, Module.algebraMap_end_apply, Module.algebraMap_end_apply]
+  repeat rw [← CompatibleSMul.map_smul, smul'_mk, ← Submonoid.mk_smul, mk_cancel]
+  rw [coe_comp, coe_coe, Function.comp_apply, iso_apply, fromLocalizedModule'_mk_one, map_apply]
+  have one : (isUnit_one (M := Module.End R (LocalizedModule S M₁))).unit = 1 := by ext; simp
```

Comment:
This can be a separate lemma, in the appropriate file.

Reply:
I cleaned up that proof a bit. I found `Module.End_isUnit_iff`, which handles the injectivity. I also added a simp theorem `IsUnit.unit_one` that simplifies `(IsUnit 1).unit`, so now the simplifier can fully simplify `fromLocalizedModule' S f (LocalizedModule.mk m 1)` without needing to add the new lemma `fromLocalizedModule'_mk_one`.